man/fi_getinfo.3.md

```diff
@@ -584,6 +584,11 @@ supported set of modes will be returned in the info structure(s).
   receive buffer at the target.  This is true even for operations that would
   normally not consume posted receive buffers, such as RMA write operations.
 
+*FI_NO_ANY_SOURCE*
+: This mode bit only applies to receive operations, including tagged receives,
+  with FI_DIRECTED_RECV capability enabled.  When set, the src_addr parameter
+  cannot be set to FI_ADDR_UNSPEC when posting a receive buffer.
```

Comment:
tag format only applies to tags. This applies to the `src_addr`.

Reply:
They end up being related.  FI_TAG_MPI needs any source.  FI_TAG_CCL does not.

Even if kept separate, this is a proposing a complex interaction, where the app sets directed_recv, app sets tag format, but the provider indicates if any source is allowed.  The latter is better as an application request, not a provider restriction.

Whether we want at least 8 possible combinations here that a provider needs to support or more limited set is open.  In the past, merging separate features has caused problems.

Reply:
Then we need to update the tag format man page to say that FI_TAG_CCL also disallow application posting recv with FI_ADDR_UNSPEC as src_addr. 

However, having separate control may still be necessary (maybe not mode bit though). MPI has user control option to indicate that no anysource receive is going to be used so that optimization can be done by the implementation. Such case can't use FI_TAG_CCL.

Reply:
I would switch to a separate cap bit, something like FI_SRC_DIRECTED_RECV, that disallows any source

Reply:
Would an endpoint option be a better choice? e.g. FI_OPT_ANY_SOURCE_PERMMITED. Application can set it to false to disallow anysource receives.

Reply:
It could work, but it requires the provider to allocate an endpoint capable of supporting any source, which may be inconvenient.  I guess as long as it must be set prior to enabling the EP, the provider can work around internal implementation limits.

Conceptually, receive buffers support wildcard source for matching by default.  FI_DIRECTED_RECV extends that to support specific source matching.  The intent here is to support only specific source matching.  To me, this seems aligned with FI_DIRECTED_RECV.

Think of it similar to how FI_MSG or FI_TAGGED may be modified by FI_SEND and FI_RECV, or FI_RMA by FI_REMOTE_READ and FI_REMOTE_WRITE.

Reply:
Good, let's take the route of defining a capability bit acting as a modifier to FI_DIRECTED_RECV.